src/typechecker.rs

```diff
@@ -453,8 +455,10 @@ impl IntegerConstant {
         let new_constant = match self {
             IntegerConstant::Signed(value) => match type_id {
                 I8_TYPE_ID => NumericConstant::I8(*value as i8),
+                CCHAR_TYPE_ID => NumericConstant::I8(*value as i8),
```

Comment:
Likewise, and might need a conditional based on `c_int`'s actual size

Reply:
I changed this one to work with the amount of bits and the signedness of the value instead of the individual types. This does not add any additional logic for the cchar, cint, and usize types.